atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+        force_calculation_parameters = self.parameter_info.get('force_calculations')
+        if force_calculation_parameters is None:
+            return
+
+        sec_force_calculations = sec_force_field.m_create(ForceCalculations)
+        sec_neighbor_searching = sec_force_calculations.m_create(NeighborSearching)
+
+        for key, val in force_calculation_parameters.items():
+            if not isinstance(val, dict):
+                key = self.check_metainfo_for_key_and_Enum(ForceCalculations, key, val)
+                setattr(sec_force_calculations, key, val)
+            else:
+                if key == 'neighbor_searching':
+                    for neigh_key, neigh_val in val.items():
+                        neigh_key = self.check_metainfo_for_key_and_Enum(NeighborSearching, neigh_key, neigh_val)
+                        setattr(sec_neighbor_searching, neigh_key, neigh_val)
+                else:
+                    self.logger.warning(key + 'is not a valid force calculations section. Corresponding parameters will not be stored.')
+
+    def parse_workflow(self):
+
+        workflow_parameters = self.parameter_info.get('workflow').get('molecular_dynamics')
+        # TODO should store parameters that do not match the enum vals as x_h5MD params, not sure how with MDParser??
+        if workflow_parameters is None:
+            return
+
+        def get_workflow_results(property_type_dict, observables, workflow_results):
+            property_key = property_type_dict['property_type_key']
+            property_value_key = property_type_dict['property_type_value_key']
+            for observable_type, observable_dict in observables.items():
+                flag_known_property = False
+                if observable_type in property_type_dict['properties_known']:
+                    property_key = observable_type
+                    property_value_key = property_type_dict['properties_known'][observable_type]
+                    flag_known_property = True
+                workflow_results[property_key] = []
+                property_dict = {property_value_key: []}
+                property_dict['label'] = observable_type
+                for key, observable in observable_dict.items():
+                    property_values_dict = {'label': key}
+                    for quant_name, val in observable.items():
+                        if quant_name == 'val':
+                            continue
+                        if quant_name == 'bins':
+                            continue
+                        if quant_name in property_type_dict['property_keys_list']:
+                            property_dict[quant_name] = val
+                        if quant_name in property_type_dict['property_value_keys_list']:
+                            property_values_dict[quant_name] = val
+                        # TODO Still need to add custom values here.
+
+                    val = observable.get('value')
+                    if val is not None:
```

Comment:
for l782-720, can we not define a common function for both value and bins? I do not quite like the repeated code.

Reply:
yes, of course. done.